[55628] Android: add logging to submission flow

MIKKH opened the pull request.

nc-kano commented on NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/CountriesConsentActivity.cs:
```diff
@@ -36,6 +38,18 @@ protected override void OnCreate(Bundle savedInstanceState)
+        protected override void OnPause()
+        {
+            base.OnPause();
+            LogUtils.LogMessage(LogSeverity.INFO, "The user is leaving Consent 3 page", null, GetCorrelationId());
```
"Leaving" logs in OnPause is not a good solution because it is called also on page change. If you have access look at Danish app at MainApplication class at OnActivityStopped method.

MIKKH replied:
I have removed logging from the OnPause() methods, and instead mirrored the implementation from the Danish app; logging the event that the user leaves an activity in the 'OnActivityStopped' method in 'MainApplication' class.

nc-kano commented on NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/LoadingPageActivity.cs:
```diff
@@ -69,15 +76,23 @@ void OnActivityFinished()
+                {
+                    LogUtils.LogMessage(
+                        LogSeverity.INFO,
+                        "Something went wrong during key sharing (INFO with correlation id)",
```
remove the "(INFO with correlation id)" it is a bad legacy code

MIKKH replied:
Has been removed.

nc-kano commented:
Code is still missing "user has closed the app", "user experienced crash" logs

MIKKH commented:
> Code is still missing "user has closed the app", "user experienced crash" logs

I have added logging of the events 'user experienced crash' and 'user has closed the app'.

For the event 'user has closed the app', I have mirrored the implementation from the Danish app; i.e. added a new service 'RemovedFromRecentDetectorService', which is used to detect when the app is closed.
